### PR TITLE
TEC-709 - Make secret optional and add a check for the secret when pr…

### PR DESCRIPTION
**Pull request by JustinGuedes**
…oceeding to get the decrypted user data

### CODE REVIEW

**Comment by civictechuser** [comment]
## Current coverage for CivicConnect.framework is `78.72%`
Files changed | - | - 
--- | --- | ---
ConnectButton.swift | `43.38%` | :no_entry_sign:
ConnectError.swift | `55.56%` | :warning:
ConnectBundle.swift | `92.59%` | :white_check_mark:
DefaultConnectSession.swift | `93.28%` | :white_check_mark:
Connect.swift | `100.00%` | :white_check_mark:
DefaultConnectSessionProvider.swift | `100.00%` | :white_check_mark:

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by JustinGuedes** [comment]
Reason for making the secret optional, is because the secret is not required when we just need to retrieve the JWT token. Also removed it from being loaded from the `Info.plist` because it can be a security issue since we are able to decompile the app and view the `Info.plist`.

File: CivicConnect/Classes/Core/Connect.swift [hunk 1]
```diff
@@ -82,12 +75,13 @@ public class Connect: NSObject {
...
     /// - Parameter bundle: A bundle that provides the application, mobile identifiers.
+    /// - Parameter secret: The secret provided from the integration portal. Optional if you just require the
+    ///                     JWT token.
     /// - Throws: Either a ConnectError.cannotFindApplicationId or ConnectError.cannotFindBundleId depending on
     ///           which field it cannot find.
-    @objc public static func initialize(withBundle bundle: ConnectBundle) throws -> Connect {
+    @objc public static func initialize(withBundle bundle: ConnectBundle, secret: String?) throws -> Connect {
         let applicationIdentifier = try getApplicationIdentifierFrom(bundle: bundle)
         let mobileApplicationIdentifier = try getMobileApplicationIdentifierFrom(bundle: bundle)
-        let secret = try getSecretFrom(bundle: bundle)
```
Comment (leolangen) [inline CivicConnect/Classes/Core/Connect.swift]: This looks like a breaking change to me. If someone is using the previous version, and they keep the secret in the info.plist, the code will no longer work as before. Is it ok to make a breaking change like this?

Comment (JustinGuedes) [reply CivicConnect/Classes/Core/Connect.swift]: Made the current Initialize method deprecated. This way it won't break for current users.

**Pull request merged**